lib/iris/cube.py

```diff
@@ -1631,9 +1636,10 @@ def lazy_data(self, array=None):
             A lazy array, representing the Cube data array.
 
         """
+        result = None
```

Comment:
don't allow this, if there is not _numpy_array and no _dask_array raise an exception